components/HeaderAppBar/index.tsx

```diff
@@ -73,7 +79,7 @@ export default function HeaderAppBar(props: HeaderAppBarProps) {
             onClose={handleCloseUserMenu}
           >
             {settings.map((setting) => (
-              <MenuItem key={setting} onClick={handleCloseUserMenu}>
+              <MenuItem key={setting} onClick={onSignOut}>
```

Comment:
just something to note: in the future if this menu gets new additions, the way this is implemented wont work. this is fine for now, just something to remember